Clother/View Models/ClothesViewModel.swift

```diff
@@ -26,59 +26,120 @@ class ClothesViewModel {
+                continue
+            }
+            if let fabric = fabric, item.fabric != fabric {
+                continue
+            }
+            
+            switch item.type {
+            case .top:
+                filteredTops.append(item)
+            case .bottom:
+                filteredBottoms.append(item)
+            case .shoes:
+                filteredShoes.append(item)
+            case .accessories:
+                filteredAccessories.append(item)
+            default:
+                continue
+            }
+        }
+        guard !filteredTops.isEmpty, !filteredBottoms.isEmpty, !filteredShoes.isEmpty, !filteredAccessories.isEmpty else {
+            return nil
+        }
+        let randomTop = filteredTops.randomElement()!
+        let randomBottom = filteredBottoms.randomElement()!
+        let randomShoe = filteredShoes.randomElement()!
+        
+        var randomAccessories: [ClothingItem] = []
+        for _ in 0..<min(3, filteredAccessories.count) {
+            let randomAccessory = filteredAccessories.randomElement()!
+            randomAccessories.append(randomAccessory)
+        }
+        
+        return (randomTop, randomBottom, randomShoe, randomAccessories)
     }
     
     
     // TODO: Saves the outfit to the necessary collections.
     func saveOutfit(item: OutfitItem) {
-        
+        userOutfits.append(item)
+        outfitsOnDisplay = userOutfits
     }
     
     // TODO: Manipulate the closet items on display such that only items with the provided input in their name are visible.
     func search(input: String) {
-        
+        clothesOnDisplay = userClothes.filter {$0.name.lowercased().contains(input.lowercased())}
     }
     
     // TODO: Manipulate the clothing items on display such that only items with the provided filters are visible.
     /*
      If a given parameter is nil, it means that no filter was applied for it.
      */
     func filterClothes(by filter: Filter) {
-        
+        clothesOnDisplay = userClothes.filter { clothingItem in
+            let typeMatch = filter.type == nil || clothingItem.type == filter.type
+            let sizeMatch = filter.size == nil || clothingItem.size == filter.size
+            let colorMatch = filter.color == nil || clothingItem.color == filter.color
+            let weatherMatch = filter.weather == nil || clothingItem.weather == filter.weather
+            let fabricMatch = filter.fabric == nil || clothingItem.fabric == filter.fabric
+            
+            return typeMatch && sizeMatch && colorMatch && weatherMatch && fabricMatch
+        }
     }
     
     // TODO: Manipulate the outfit items on display such that only items with the provided filters are visible.
     /*
      If a given parameter is nil, it means that no filter was applied for it.
      */
     func filterOutfits(topFilter: Filter, bottomFilter: Filter, shoesFilter: Filter) {
+        outfitsOnDisplay = userOutfits.filter { outfitItem in
```

Comment:
You must check all filter properties. In this implementation you have only compared the 'type' fields.